tfhe/src/core_crypto/commons/math/torus/mod.rs

```diff
@@ -83,26 +78,45 @@ macro_rules! implement {
                 return signed.cast_into();
             }
             #[inline]
-            // WARNING from documentation reproduced here
-            // It is the caller's reponsibility to provide a custom_modulus that is a safe
-            // approximation of the integer modulus they want to use in the floating point
-            // domain.
-            //
-            // If the approximate floating point modulus is too big then some values will be out of
-            // the proper range for the given integer modulus.
-            fn from_torus_custom_modulus(input: F, custom_modulus: F) -> Self {
+            fn from_torus_custom_mod(input: F, custom_modulus: Self) -> Self {
+                // Reasonging with F = f64
+                //
+                // Here we convert the custom modulus to f64 but rounding down to the closest
+                // representable value by f64. This allows to make sure that the value that
+                // is outputed is in the correct range with respect to our modulus.
+                // The max error for a u64 is 2048, i.e. max(diff(floor(u64 as f64) as u64, u64)) <
+                // 2048 This small error should be acceptable for values produced by the
+                // gaussian generation as they have to be related to the modulus itself,
+                // 2048 / 2^63 = 2^-52, basically the f64 relative error is the max relative
+                // error you can get. This is valid for other moduli >= 2^53 with a max
+                // absolute error as low as 1 e.g. for 2^64 - 2^32 + 1, which rounds down to
+                // 2^64 - 2^32 in the float domain.
+                let custom_modulus_float: F =
+                    convert_unsigned_integer_to_float_truncate(custom_modulus);
```

Comment:
May be safer to do a `round_towards_infinity`
We could run some tests to see the impact

Reply:
yep as discussed I'll check with a quick python script for moduli > 2^53 to see the impact of the product, but as now the modular reduction is done with integer the impact is probably negligible, will keep you posted

Reply:
https://github.com/zama-ai/tfhe-rs-internal/issues/326 following the discussion from yesterday